backend/functions/tests/flashcard.test.js

```diff
@@ -32,115 +36,255 @@ describe("addFlashcardSet", () => {
+        category: "Updated category",
+        cards: [
+          {question: "Updated Question 1", answer: "Updated Answer 1"},
+          {question: "Updated Question 2", answer: "Updated Answer 2"},
+          {question: "New Question", answer: "New Answer"},
+        ],
+      };
+
+      // Create a test flashcard set
+      const addFlashcardSet = fft.wrap(myFunctions.addFlashcardSet);
+      const addData = {
+        title: "Test Flashcard Set",
+        category: "Test Category",
+        cards: [
+          {question: "Test Question 1", answer: "Test Answer 1"},
+          {question: "Test Question 2", answer: "Test Answer 2"},
+        ],
+      };
+
+      const context = {auth: {uid: user.uid}};
+
+      const addRes = await addFlashcardSet(addData, context);
+      docsToDelete.push(addRes.flashcardId);
+
+      // Validate reference to flashcard set is on "user" document
+
+      expect(addRes.flashcardId).toBeTruthy();
+
+      const res = await updateFlashcardSet(
+          {flashcardId: addRes.flashcardId, ...data},
+          context,
+      );
+
+      // Validate response from updating flashcard set
+      expect(res.flashcardId).toEqual(addRes.flashcardId);
+      expect(res.title).toEqual(data.title);
+      expect(res.category).toEqual(data.category);
+      expect(res.cards.length).toEqual(3);
+      expect(res.cards[0].question).toEqual(data.cards[0].question);
+      expect(res.cards[0].answer).toEqual(data.cards[0].answer);
+      expect(res.cards[1].question).toEqual(data.cards[1].question);
+      expect(res.cards[1].answer).toEqual(data.cards[1].answer);
+      expect(res.cards[2].question).toEqual(data.cards[2].question);
+      expect(res.cards[2].answer).toEqual(data.cards[2].answer);
+      expect(res.creatorId).toBe(context.auth.uid);
+      expect(res.timestamp).toBeTruthy();
+    });
+
+    test("throws error if not authenticated", async () => {
+      const updateFlashcardSet = fft.wrap(myFunctions.updateFlashcardSet);
+      await expect(updateFlashcardSet({}, {})).rejects.toEqual(
+          new Error("You must be authenticated to update a flashcard set."),
+      );
+    });
+
+    test("throws error if missing fields", async () => {
+      const updateFlashcardSet = fft.wrap(myFunctions.updateFlashcardSet);
+      const context = {auth: {uid: user.uid}};
+      const cases = [
+        ["flashcardId", {}],
+        ["title", {flashcardId: "flashcardId"}],
+        ["category", {flashcardId: "flashcardId", title: "title"}],
+        ["cards", {flashcardId: "flashcardId", title: "title", category: "category"}],
+      ];
+
+      for (const testVal of cases) {
+        await expect(updateFlashcardSet(testVal[1], context)).rejects.toEqual(
+            new Error(`The "${testVal[0]}" field must be provided.`),
+        );
+      }
+    });
+
+    test("throws error if title & category fails pre-condition", async () => {
+      const updateFlashcardSet = fft.wrap(myFunctions.updateFlashcardSet);
```

Comment:
I think we can exclude the tests for:
- `"throws error if title & category fails pre-condition"`
- `"throws error if cards fail pre-condition"`

Since it's pretty much duplicated from the previous test with `addFlashcardSet` (since they use the same validation scheme).
- This has the added benefit of removing that duplicated code linter error